Update the Next.js Pages guide to use the latest tooling (#13001)
cloudflare/cloudflare-docs: content/pages/framework-guides/nextjs/deploy-a-nextjs-site.md

## Configure and deploy a project without C3

> If you already have a Next.js project or wish to manually create and deploy one without using C3, Cloudflare recommends that you use `@cloudflare/next-on-pages` and refer to its [README](https://github.com/cloudflare/next-on-pages/blob/3846730c4a0d12/packages/next-on-pages#cloudflarenext-on-pages) for instructions and additional information to help you develop and deploy your project.

Reviewer: @dario-piotrowicz I recommend against this as the link will take people to a stale version of the doc over time. What I recommend instead is that you create a manual anchor in the readme.md via `<a name="manual-installation">` for example and refer to it here. Manual anchors are unlikely to break easily so you get both the stability of the link as ability to refresh the readme.md content without having to come back here to update the link.

> If you already have a Next.js project or wish to manually create and deploy one without using C3, Cloudflare recommends that you use `@cloudflare/next-on-pages` and refer to its [README](https://github.com/cloudflare/next-on-pages/blob/3846730c4a0d12/packages/next-on-pages#cloudflarenext-on-pages) for instructions and additional information to help you develop and deploy your project.

Author: The `#cloudflarenext-on-pages` is actually unnecessary, it is pointing to the readme title (and all it does is making github scroll the user to the beginning of the README), if you're ok with it I might as well just remove it and simply point to the file 🙂

{{</Aside>}}

> To set up bindings for use in local development, you will use the `setupDevPlatform` function provided by [`@cloudflare/next-on-pages/next-dev`](https://github.com/cloudflare/next-on-pages/blob/3846730c4a0d12/internal-packages/next-dev). This function, tries to read the project's [`wrangler.toml`](/workers/wrangler/configuration/) file and based on it sets up a platform emulation that your Next.js application can make use of locally.

Reviewer: same here

Author: Here it is already pointing to the readme itself, so if it's ok by you I'll just revert the `blob/` part to `/tree/main` 🙂

```js
---
filename: next.config.mjs
highlight: [1-7]
---
import { setupDevPlatform } from '@cloudflare/next-on-pages/next-dev'

> // note: the if statement is present because we

Author: ```suggestion
// note: the if statement is present because you
```

```js
---
filename: next.config.js / next.config.cjs
highlight: [1-6]
---
// note: the if statement is because we only need to use the function

> // during development (the check is recommended but completely optional)

Reviewer: Why do you mention that the check is optional? I find that part confusing. Is there a valid workflow when you'd want to invoke `setupDevPlatform` without the guard? I suggest you drop the note in parenthesis.

Author: I just feel like developers can try to avoid extra code when necessary, so it feels nice to me to let people know that they do not have to check `process.env.NODE_ENV` if they don't want to (instead of giving the impression that they absolutely need to)

It's not a big deal and it is only a single link of code

I'm ok removing it if you prefer 🙂

Author: removed 🙂👍

if (process.env.NODE_ENV === "development") {
  const { setupDevPlatform } = require("@cloudflare/next-on-pages/next-dev")
  setupDevPlatform()
}

/** @type {import('next').NextConfig} */
const nextConfig = {}

module.exports = nextConfig
```

{{</tab>}}
{{</tabs>}}

Also make sure to have a `wrangler.toml` file at the root of your project with a declaration for a KV binding named `MY_KV`:

```toml
---
filename: wrangler.toml
highlight: [5-7]
---
name = "my-next-app"

compatibility_flags = ["nodejs_compat"]

[[kv_namespaces]]
binding = "MY_KV"

> id = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"

Reviewer 2: ```suggestion
id = "<YOUR_KV_NAMESPACE_ID>"
```

```

### Set up bindings for a deployed application

To access bindings in a deployed application, you will need to [configure](/pages/functions/bindings/) any necessary bindings and connect them to your project via your project's settings page in the Cloudflare dashboard.

### Add bindings to Typescript projects

> If your project is using typescript you will want to set up proper type support so that you can access
> your bindings in a type safe and convenient manner.

Reviewer 2: ```suggestion
If your project is using TypeScript, you will want to set up proper type support so you can access your bindings in a type-safe and convenient manner.
```

To get proper type support, you need to create a new `env.d.ts` file in your project and extend the `CloudflareEnv` (used by `getRequestContext`) interface with your [bindings](/pages/functions/bindings/).

{{<Aside type="note">}}

> Note: Projects created with C3 have a default `env.d.ts` file.

Reviewer 2: ```suggestion
Projects created with C3 have a default `env.d.ts` file.
```

And preview your project by running:

```sh
$ npx wrangler pages dev .vercel/output/static
```

{{<Aside type="note">}}

> The `wrangler pages dev` command needs to run the application using the [`nodejs_compat`](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) compatibility flag, that can be either specified in the `wrangler.toml` file or provided to the command as an inline argument: `--compatibility-flag=nodejs_compat`.

Reviewer 2: ```suggestion
The [`wrangler pages dev`](/workers/wrangler/commands/#dev-1) command needs to run the application using the [`nodejs_compat`](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) compatibility flag. The `nodejs_compat` flag can be specified in either your project's `wrangler.toml` file or provided to the command as an inline argument: `--compatibility-flag=nodejs_compat`.
```
